ui/appui-react/src/appui-react/widgets/WidgetDef.tsx

```diff
@@ -213,7 +213,15 @@ export class WidgetDef {
 
     this.setCanPopout(config.canPopout);
 
-    const canFloat = config.canFloat;
+    let canFloat = config.canFloat;
+    if (config.allowedPanels && config.allowedPanels.length === 0) {
+      if (config.canFloat !== true) {
+        canFloat = config.canFloat;
+      } else {
+        canFloat = true;
+      }
+    }
```

Comment:
The logic appears to be backwards here, this code says: if we have `allowedPanels` an empty array, and `canFloat` is not `true`, leave it as is, otherwise (it is already `true`, force it to `true) ?

Is this setting even read anywhere if your tests worked ?

Reply:
config.canFloat can also be an object that is the container id of the floating object. So the logic is as this, if we have an empty allowed panels array check if the config.canFloat is not a true boolean. If its false, we want to set it to true anyway, but if a config.canFloat object was passed in, we want to set can float to that

Reply:
Ok, then there is still an issue, because `false` or `undefined` will use the `canFloat !== true` true branch. The test should probably `typeof  config.canFloat === "object"`

Reply:
Currently, the documented behavior is that an empty allowedPanels array means that all the panels are allowed. This was for backward compatibility.

Reply:
Yeah I see what you mean now, on it

Reply:
> Currently, the documented behavior is that an empty allowedPanels array means that all the panels are allowed. This was for backward compatibility.



> Currently, the documented behavior is that an empty allowedPanels array means that all the panels are allowed. This was for backward compatibility.

Passing in nothing at all so that it is undefined means that all panels are allowed now. Whilst passing in a blank array will restrict docking. Does something need to be changed to account for backwards compatibility? Or does the documentation need changed further? @NancyMcCallB

Reply:
> Ok, then there is still an issue, because `false` or `undefined` will use the `canFloat !== true` true branch. The test should probably `typeof config.canFloat === "object"`

This logic has been fixed